src/js/util/param_spec.js

```diff
@@ -0,0 +1,40 @@
+function ParamSpec(sourceLow, sourceHigh, distLow, distHigh, clamp = true) {
```

Comment:
I was thinking of this as mapping or specifying a range of values for a given parameter. Maybe we could call it `RangeMapper`?

> map/unmap aren't really clear function names to me?

if we called it `RangeMapper` would map/unmap work? names are hard.